akka-actor/src/main/scala/akka/event/Logging.scala

```diff
@@ -487,6 +487,8 @@ object Logging {
     case WarningLevel ⇒ classOf[Warning]
     case InfoLevel    ⇒ classOf[Info]
     case DebugLevel   ⇒ classOf[Debug]
+    case unsupported  ⇒ throw new LoggerInitializationException(s"Unsupported Logger type $unsupported.")
```

Comment:
I'd prefer `IllegalArgumentException` and it's s"Unsupported log level [$level]".
An convention we try to use is to enclose parameters like that in `[ ]` in exceptions and error messages. Reason is that it's easier to see that it's a dynamic value and e.g. empty string wouldn't be visible otherwise.

Reply:
similar in several other places below